recipes/gstreamer/all/conanfile.py

```diff
@@ -55,156 +58,199 @@ def config_options(self):
             # https://gitlab.freedesktop.org/gstreamer/gst-build/-/issues/133
-            raise ConanInvalidConfiguration("shared GStreamer cannot link to static GLib")
+            raise ConanInvalidConfiguration(
+                "shared GStreamer cannot link to static GLib")
 
     def build_requirements(self):
-        self.build_requires("meson/0.61.2")
-        self.build_requires("pkgconf/1.7.4")
+        self.tool_requires("meson/0.63.3")
+        self.tool_requires("cmake/3.24.2")
+        self.tool_requires("pkgconf/1.9.3")
         if self.options.with_introspection:
-            self.build_requires("gobject-introspection/1.70.0")
+            self.tool_requires("gobject-introspection/1.72.0")
         if self.settings.os == 'Windows':
-            self.build_requires("winflexbison/2.5.24")
+            self.tool_requires("winflexbison/2.5.24")
         else:
-            self.build_requires("bison/3.7.6")
-            self.build_requires("flex/2.6.4")
+            self.tool_requires("bison/3.8.2")
+            self.tool_requires("flex/2.6.4")
+
+    def layout(self):
+        basic_layout(self, src_folder="source")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("%s-%s" % (self.name, self.version), self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    def _configure_meson(self):
-        if self._meson:
-            return self._meson
-        meson = Meson(self)
-        if self._is_msvc:
-            if tools.Version(self.settings.compiler.version) < "14":
-                meson.options["c_args"] = " -Dsnprintf=_snprintf"
-                meson.options["cpp_args"] = " -Dsnprintf=_snprintf"
+    def generate(self):
+        tool_chain = MesonToolchain(self)
+        if is_msvc(self):
+            if Version(self.settings.compiler.version) < "14":
+                tool_chain.project_options["c_args"] = " -Dsnprintf=_snprintf"
+                tool_chain.project_options["cpp_args"] = " -Dsnprintf=_snprintf"
         if self.settings.get_safe("compiler.runtime"):
-            meson.options["b_vscrt"] = str(self.settings.compiler.runtime).lower()
-        meson.options["tools"] = "disabled"
-        meson.options["examples"] = "disabled"
-        meson.options["benchmarks"] = "disabled"
-        meson.options["tests"] = "disabled"
-        meson.options["introspection"] = "enabled" if self.options.with_introspection else "disabled"
-        meson.configure(build_folder=self._build_subfolder,
-                        source_folder=self._source_subfolder,
-                        args=['--wrap-mode=nofallback'])
-        self._meson = meson
-        return self._meson
+            tool_chain.project_options["b_vscrt"] = str(
+                self.settings.compiler.runtime).lower()
+        tool_chain.project_options["tools"] = "enabled" if self.options.with_tools else "disabled"
```

Comment:
Tools and benchmarks are not options available on the main project and meson will fail:
https://gitlab.freedesktop.org/gstreamer/gstreamer/-/blob/1.20.4/meson_options.txt

I have tried with "gstreamer:tools" = "enabled/disabled", but it doesn't seem to work.